src/LodeRunner.Core/Models/BasePayload.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LodeRunner.Core.Models
+{
+    /// <summary>
+    /// Base class for payload.
+    /// </summary>
+    public class BasePayload : INotifyPropertyChanged
+    {
+        /// <summary>
+        /// Occurs when a property value changes.
+        /// </summary>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Gets or sets the properties changed list.
+        /// </summary>
+        /// <value>
+        /// The changed properties.
+        /// </value>
+        public List<string> PropertiesChanged { get; set; } = new ();
+
+        /// <summary>
+        /// Called when [property changed].
+        /// </summary>
+        /// <param name="fieldName">The field name.</param>
+        protected void OnPropertyChanged([CallerMemberName] string fieldName = null)

```

Comment:
Why do we need this?  I didn't see any subscriptions to this event in cursory look.  

If it is to minimize mapping work I don't think it will be effective as the mapper uses reflection and the reason to minimize mapping would be to reduce the reflection, but once we introduce the `SetField()` method which uses reflection it is causing the code to use costly reflection on every value assignment to the field.

Reply:
@gortegaMS

Reply:
Refactor BasePayload and updated DataDictionary